tests/baselines/reference/mappedTypeRelationships.errors.txt

```diff
@@ -1,39 +1,71 @@
+        Type 'T[keyof U]' is not assignable to type 'U[string]'.
+          Type 'T[string]' is not assignable to type 'U[string]'.
+            Type 'T' is not assignable to type 'U'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(22,12): error TS2536: Type 'keyof U' cannot be used to index type 'T'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(26,5): error TS2536: Type 'K' cannot be used to index type 'T'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(27,5): error TS2322: Type 'T[K]' is not assignable to type 'U[K]'.
   Type 'T[string]' is not assignable to type 'U[K]'.
-    Type 'T' is not assignable to type 'U'.
+    Type 'T[string]' is not assignable to type 'U[string]'.
+      Type 'T' is not assignable to type 'U'.
+        Type 'T[K]' is not assignable to type 'U[string]'.
+          Type 'T[string]' is not assignable to type 'U[string]'.
+            Type 'T' is not assignable to type 'U'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(27,12): error TS2536: Type 'K' cannot be used to index type 'T'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(31,5): error TS2322: Type 'T[keyof T] | undefined' is not assignable to type 'T[keyof T]'.
   Type 'undefined' is not assignable to type 'T[keyof T]'.
+    Type 'undefined' is not assignable to type 'T[string]'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(36,5): error TS2322: Type 'T[K] | undefined' is not assignable to type 'T[K]'.
   Type 'undefined' is not assignable to type 'T[K]'.
+    Type 'undefined' is not assignable to type 'T[string]'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(41,5): error TS2322: Type 'U[keyof T] | undefined' is not assignable to type 'T[keyof T]'.
   Type 'undefined' is not assignable to type 'T[keyof T]'.
+    Type 'undefined' is not assignable to type 'T[string]'.
 tests/cases/conformance/types/mapped/mappedTypeRelationships.ts(42,5): error TS2322: Type 'T[keyof T]' is not assignable to type 'U[keyof T] | undefined'.
   Type 'T[string]' is not assignable to type 'U[keyof T] | undefined'.
     Type 'T[string]' is not assignable to type 'U[keyof T]'.
-      Type 'T[keyof T]' is not assignable to type 'U[keyof T]'.
-        Type 'T[string]' is not assignable to type 'U[keyof T]'.
-          Type 'T' is not assignable to type 'U'.
+      Type 'T[string]' is not assignable to type 'U[string]'.
+        Type 'T' is not assignable to type 'U'.
+          Type 'T[keyof T]' is not assignable to type 'U[keyof T]'.
+            Type 'T[string]' is not assignable to type 'U[keyof T]'.
+              Type 'T[string]' is not assignable to type 'U[string]'.
+                Type 'T' is not assignable to type 'U'.
+                  Type 'T[keyof T]' is not assignable to type 'U[string]'.
+                    Type 'T[string]' is not assignable to type 'U[string]'.
+                      Type 'T' is not assignable to type 'U'.

```

Comment:
Is there any way to not do this? Going around in circles here? ` 'T' is not assignable to type 'U'.` because ` 'T' is not assignable to type 'U'.` because ` 'T' is not assignable to type 'U'.`